src/app/effects/settings.ts

```diff
@@ -8,7 +8,6 @@ import { Actions, Effect } from '@ngrx/effects';
 import { Store } from '@ngrx/store';
 
 import { LocalStorageService } from '../services/localstorage.service';
-import { Settings } from '../models/settings';
```

Comment:
there is no need to reference Settings Model here - the object is created 'on the fly' here : 

`Object.assign({}, store.settings, action.payload`

-our base is a totally empty object
-empty object is set equal to the current Settings Object in the store (which is of type Settings, as defined in /models/settings)
-the action.payload's type is of type State, which extends the Settings type, adding the 'complete' field